src/tools/compiletest/src/directive-list.rs

```diff
@@ -117,6 +117,7 @@ const KNOWN_DIRECTIVE_NAMES: &[&str] = &[
     "incremental",
     "known-bug",
     "llvm-cov-flags",
+    "max-llvm-major-version",
```

Comment:
Because usually we're interested in changes between **major** LLVM versions (LLVM is semver). For example, for FileCheck-based tests usually we are interested to limit the test on a maximum major LLVM version e.g. 19 (but we still want the test to run on LLVM 19.1, 19.99.99).

Reply:
If you look at the assembly tests, this is exactly the property they want: to run tests up to a maximum llvm major version.

Alternatively, we could also offer `//@ llvm-version-less-than`.

Reply:
It seems quite weird that we have `min-llvm-version` which supports semver and we are adding `max-llvm-major-version` which only looks for the major part. It seems like after this, we will need to add another directive when we need to check minor versions? I got the point but it doesn't seem to justify the situation IMO.

Reply:
> It seems like after this, we will need to add another directive when we need to check minor versions? I got the point but it doesn't seem to justify the situation IMO.

Such a need for minor version checking has not yet materialized AFAIK. We can always update the directives to better support our use cases if such a need does occur, but needing to constrain on a major llvm version is a problem at present, especially between llvm major version bumps.

Reply:
So why aren’t we doing this now? Just curious, what’s the reason? The cost for doing it doesn't seem much, or am I missing something? It really seems odd that the LLVM versioning interface for min and and directives is entirely different. Wouldn't it be better when they are unified and documented as {min/max}-llvm-version?

Reply:
max-llvm-version would be pretty awkward to use, because you'd always have to write it as `max-llvm-version: 19.99.99` or so.

Reply:
No, you can still just write the major part.

Reply:
But wouldn't `max-llvm-version: 19` exclude `19.1.0`, for example, because `19.1.0 > 19`? Or is that not how the version comparison works?

Reply:
Ah, got you now. Yeah, in order to ignore 19 completely, you will need to write 19.99.99, which seems not very good.

Reply:
Yeah I should've made that more clear. I did that exactly i.e. max-llvm-version in the initial impl, then it occurred to me it is very awkward when I tried to write the unit tests.